src/client/datascience/jupyter/kernels/kernel.ts

```diff
@@ -150,6 +162,29 @@ export class Kernel implements IKernel {
             }
         }
     }
+    private async trackNotebookCellPerceivedColdTime(
+        stopWatch: StopWatch,
+        notebookPromise: Promise<INotebook>
+    ): Promise<void> {
+        if (this.perceivedJupyterStartupTelemetryCaptured) {
+            return;
+        }
+        const notebook = await notebookPromise;
+        if (!notebook) {
+            return;
+        }
+        // Setup telemetry
+        if (!this.perceivedJupyterStartupTelemetryCaptured) {
+            this.perceivedJupyterStartupTelemetryCaptured = true;
+            sendTelemetryEvent(Telemetry.PerceivedJupyterStartupNotebook, stopWatch?.elapsedTime);
+            const disposable = notebook.onSessionStatusChanged((e) => {
+                if (e === ServerStatus.Busy) {
```

Comment:
Name of this function is throwing me a bit. Is it perceived cold time to execute a cell? It looks like it's firing on changing of status to busy, which would just be pre-execute startup time at least as I see it. Not sure if it's a code issue or if I'm not getting what it's trying to measure.

Reply:
> s it perceived cold time to execute a cell?

Yes

> oks like it's firing on changing of status to busy, which would just be pre-execute startup time at least as I see i

yes, borrowed existing logic from the `ineractiveBase.ts` file for the same funciotnality, see here 
https://github.com/microsoft/vscode-jupyter/blob/88f11c2efc7db79b8faf10f10440667db2adc8bd/src/client/datascience/interactive-common/interactiveBase.ts#L685-L685

Reply:
Sure that works. The name just implied to me that value would be from user opens a notebook and presses some type of run command to the time that the first cell completes, not when the first cell starts executing, but that may just be my understanding.

Reply:
I have fixed this to just want for the promise to complete, that will be a more accurate measure of when execution completes.